pubsub/awspubsub/awspubsub.go

```diff
@@ -135,26 +143,24 @@ func (t *topic) SendBatch(ctx context.Context, dms []*driver.Message) error {
 		select {
 		case <-ctx.Done():
 			errc <- ctx.Err()
-			break
-		case sem <- struct{}{}:
+		case t.sem <- struct{}{}:
```

Comment:
How about 
```
return ctx.Err()
```
on the above line? When the context is done, you want to terminate immediately.

Reply:
I don't think that's correct, because it will exit the function while some goroutines that it started may still be running. I'd rather explicitly wait for all of the goroutines to exit.

Reply:
OK, I guess since they take the same context they will stop quickly.

But it seems silly to start more goroutines after the context is Done (possibly waiting for the semaphore to do so). You should find a way to return ASAP, while waiting for outstanding goroutines to finish.

Reply:
OK, now short-circuiting if ctx is done.